geo-types/src/coordinate.rs

```diff
@@ -106,16 +118,20 @@ use std::ops::{Add, Div, Mul, Neg, Sub};
 /// assert_eq!(q.x, -p.x);
 /// assert_eq!(q.y, -p.y);
 /// ```
-impl<T> Neg for Coordinate<T>
+impl<T, Z, M> Neg for CoordTZM<T, Z, M>
 where
     T: CoordNum + Neg<Output = T>,
+    Z: ZCoord + Neg<Output = Z>,
+    M: Measure + Neg<Output = M>,
 {
     type Output = Self;
 
     fn neg(self) -> Self {
         coord! {
             x: -self.x,
             y: -self.y,
+            z: -self.z,
+            m: -self.m, // todo - is this right?
```

Comment:
Great question. We should look at what other libraries do. And maybe also call the behavior out in the documentation.

Reply:
I have looked at JTS and a bit at geos -- I couldn't find similar code there. I think it won't become apparent until we begin implementing .z and .m algorithms en-mass. For now I think we might as well apply exactly the same operation to z and m as to x and y.